deepinv/datasets/div2k.py

```diff
@@ -0,0 +1,110 @@
+import os
+
+from PIL import Image
+import torch
+
+from .utils import calculate_md5_for_folder, download_zipfile, extract_zipfile
+
+
+class DIV2K(torch.utils.data.Dataset):
+
+    # https://data.vision.ee.ethz.ch/cvl/DIV2K/
+    zipfile_urls = {
+        'DIV2K_train_HR.zip' : 'http://data.vision.ee.ethz.ch/cvl/DIV2K/DIV2K_train_HR.zip',
+        'DIV2K_valid_HR.zip' : 'http://data.vision.ee.ethz.ch/cvl/DIV2K/DIV2K_valid_HR.zip',
+    }
+
+    # for integrity of downloaded data
+    checksums = {
+        'DIV2K_train_HR' : 'f9de9c251af455c1021017e61713a48b',
+        'DIV2K_valid_HR' : '542325e500b0a474c7ad18bae922da72',
+    }
+
+    def __init__(self, root: str, mode: str='train', download: bool=False, transform=None) -> None:
+        super().__init__()
+
+        self.transform = transform
+        self.root = root
+        train_dir_path = os.path.join(self.root, 'DIV2K_train_HR')
+        valid_dir_path = os.path.join(self.root, 'DIV2K_valid_HR')
+
+        # verify that dataset is available at self.root and use it
+        if self._check_dataset_exists():
+            print(f'''
+            Dataset is available at `{self.root}`.
+            `download` flag is not taken into account.
+            ''')
+        # otherwise we try to download the whole dataset
+        elif download:
+            if not os.path.isdir(self.root):
+                os.makedirs(self.root)
+            if os.path.exists(train_dir_path):
+                raise ValueError(f'''
```

Comment:
It raises an error since we know that self.root doesn't point out to a complete dataset and we are trying to download it but the train folder already exists and we don't know what to do with it (it can be corrupted or user may not want us to rewrite it)